logic.py

```diff
@@ -232,7 +232,7 @@ def constant_symbols(x):
     elif is_prop_symbol(x.op) and not x.args:
         return {x}
     else:
-        return {symbol for arg in x.args for symbol in constant_symbols(arg)}
+        return set().union(*map(constant_symbols, x.args))
```

Comment:
Same as my above review.

Reply:
If you think the comprehension is easier to understand, let's stick with that.